homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+        or (start_minutes is not None)
+        or (start_seconds is not None)
+    ):
+        matching_timers = [
+            t
+            for t in matching_timers
+            if (t.start_hours == start_hours)
+            and (t.start_minutes == start_minutes)
+            and (t.start_seconds == start_seconds)
+        ]
+        if not matching_timers:
+            # No matches
+            return matching_timers
+
+    if "device_id" not in slots:
+        # Can't re-order based on area/floor
+        return matching_timers
+
+    # Use device id to order remaining timers
+    device_id: str = slots["device_id"]["value"]
+    device = timer_manager.device_registry.async_get(device_id)
+    if (device is None) or (device.area_id is None):
+        return matching_timers
+
+    area = timer_manager.area_registry.async_get_area(device.area_id)
+    if area is None:
+        return matching_timers
+
+    def area_floor_sort(timer: TimerInfo) -> int:
+        if timer.area_id == area.id:
+            return -2
+
+        if timer.floor_id == area.floor_id:
+            return -1
+
+        return 0
+
+    matching_timers.sort(key=area_floor_sort)
+
+    return matching_timers
+
+
+def _get_total_seconds(slots: dict[str, Any]) -> int:
+    """Return the total number of seconds from hours/minutes/seconds slots."""
+    total_seconds = 0
+    if "hours" in slots:
+        total_seconds += 60 * 60 * int(slots["hours"]["value"])
+
+    if "minutes" in slots:
+        total_seconds += 60 * int(slots["minutes"]["value"])
+
+    if "seconds" in slots:
+        total_seconds += int(slots["seconds"]["value"])
+
+    return total_seconds
+
+
+class SetTimerIntentHandler(intent.IntentHandler):
+    """Intent handler for starting a new timer."""
+
+    intent_type = intent.INTENT_SET_TIMER
+    slot_schema = {
+        vol.Required(vol.Any("hours", "minutes", "seconds")): cv.positive_int,
+        vol.Optional("name"): cv.string,
+        vol.Optional("device_id"): cv.string,
+        vol.Optional("data"): dict,
+    }
+
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
+        """Handle the intent."""
+        hass = intent_obj.hass
+        timer_manager: TimerManager = hass.data[TIMER_DATA]
+        slots = self.async_validate_slots(intent_obj.slots)
+
+        device_id: str | None = None
+        if "device_id" in slots:
+            device_id = slots["device_id"]["value"]
+
+        name: str | None = None
+        if "name" in slots:
+            name = slots["name"]["value"]
+
+        hours: int | None = None
+        if "hours" in slots:
+            hours = int(slots["hours"]["value"])
+
+        minutes: int | None = None
+        if "minutes" in slots:
+            minutes = int(slots["minutes"]["value"])
+
+        seconds: int | None = None
+        if "seconds" in slots:
+            seconds = int(slots["seconds"]["value"])
+
+        data: dict[str, Any] | None = None
+        if "data" in slots:
```

Comment:
Any reason we're hardcoding it to data and not just limiting it to "follow-up command" ? That's really the only use case we have right ?

Reply:
I suggest we move supporting follow-up command to its own PR once the timers have landed. Also, we should make sure that we parse the follow-up command such that we know it is a sentence that matches.

Reply:
It should be handled by same agent doing the timer.